File: app/src/main/java/vergecurrency/vergewallet/viewmodel/PaperkeyVerificationViewModel.java
package vergecurrency.vergewallet.viewmodel;

import androidx.lifecycle.ViewModel;

import vergecurrency.vergewallet.service.model.MnemonicManager;
import vergecurrency.vergewallet.service.model.PreferencesManager;

public class PaperkeyVerificationViewModel extends ViewModel {
    private String[] seed;

    public PaperkeyVerificationViewModel() {
        seed = MnemonicManager.getMnemonicFromJSON(PreferencesManager.getMnemonic());
    }

    public String[] getSeed() {
        return seed;

    }

    public void setFirstLaunch(boolean isFirstLaunch) {
        PreferencesManager.setFirstLaunch(isFirstLaunch);
    }
}

File: app/src/main/kotlin/vergecurrency/vergewallet/viewmodel/PaperkeyVerificationViewModel.kt
package vergecurrency.vergewallet.viewmodel

import androidx.lifecycle.ViewModel
import vergecurrency.vergewallet.service.model.EncryptedPreferencesManager

import vergecurrency.vergewallet.service.model.MnemonicManager
import vergecurrency.vergewallet.service.model.PreferencesManager

class PaperkeyVerificationViewModel : ViewModel() {
    var seed: Array<CharArray> = MnemonicManager.getMnemonicFromJSON(EncryptedPreferencesManager.mnemonic!!)!!

    fun setFirstLaunch(isFirstLaunch: Boolean) {
        PreferencesManager.isFirstLaunch = isFirstLaunch
    }
}
